src/screens/Map/MapScreen.tsx

```diff
@@ -38,6 +38,18 @@ export const MapScreen = ({navigation}: MapScreenProps<'MapScreen'>) => {
       activeItemIndex: 0,
     });
   };
+  const navigateToTripSearch: TravelFromAndToLocationsCallback = (
+    location,
+    destination,
+  ) => {
+    navigation.navigate({
```

Comment:
@gorandalum I have tried the other way around as you suggested but, the parameters inside `TripSearch` is not merged.

I tried: 

```
navigation.navigate({
      name: 'Dashboard',
      params: {
        screen: 'TripSearch',
        params: {
          [destination]: location,
        },
      },
      merge: true,
    });
```
But if you leave it as is now I have noticed that the `TripSearch` is already present in the stack because now Dashboard becomes the default when the app loads. 
I didn't find any other way to get the previous parameter on that screen, so without the merge option, it always replaces the last from selection to the current position, which breaks the flow we want.

Reply:
If it can't be done then we should navigate directly. But `any` should be used even though, so the NavigationProp have to be updated so the `TripSearch` is a visible screen. I'll take a look now.

Reply:
I couldn't get it to work 🤔 Let's hear what they have to say:
https://github.com/react-navigation/react-navigation/issues/10995

Reply:
But for now just make it work with this. Try to remove `any` by using `CompositeScreenProps` so the `TripSearch` is available for the `navigation` object.